whois-nrtm4/src/main/java/net/ripe/db/nrtm4/persist/SnapshotFileRepository.java

```diff
@@ -0,0 +1,87 @@
+package net.ripe.db.nrtm4.persist;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.Optional;
+
+
+@Repository
+public class SnapshotFileRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final RowMapper<SnapshotFile> rowMapper = (rs, rowNum) ->
+        new SnapshotFile(
+            rs.getLong(1),
+            rs.getLong(2),
+            rs.getString(3),
+            rs.getString(4),
+            rs.getLong(5)
+        );
+
+    private final String snapshotFileFields = "sf.id, sf.version_id, sf.name, sf.hash, sf.created ";
+
+    @Autowired
+    public SnapshotFileRepository(@Qualifier("nrtmDataSource") final DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+    }
+
+    public SnapshotFile save(
+        final long versionId,
+        final String name,
+        final String hash
+    ) {
+        final KeyHolder keyHolder = new GeneratedKeyHolder();
+        final long now = System.currentTimeMillis();
+        jdbcTemplate.update(connection -> {
```

Comment:
Can we get the timestamp same way we get for created and lastmodified attribute for whois?

Reply:
check TimestampAttributeGenerator class

Reply:
switched to use dataTimeProvider -- same as the last and history tables in whois.